addons/mail/static/src/new/messaging.js

```diff
@@ -473,31 +473,42 @@ export class Messaging {
         this.discuss.threadId = channel.id;
     }
 
-    async getChat({ userId }) {
-        let user = this.users[userId];
-        if (!user) {
-            this.users[userId] = { id: userId };
-            user = this.users[userId];
-        }
-        if (!user.partner_id) {
-            const [userData] = await this.orm.silent.read("res.users", [user.id], ["partner_id"], {
-                context: { active_test: false },
-            });
-            if (userData) {
-                user.partner_id = userData.partner_id[0];
+    async getChat({ userId, partnerId }) {
+        if (!partnerId) {
+            let user = this.users[userId];
+            if (!user) {
+                this.users[userId] = { id: userId };
+                user = this.users[userId];
             }
+            if (!user.partner_id) {
+                const [userData] = await this.orm.silent.read(
+                    "res.users",
+                    [user.id],
+                    ["partner_id"],
+                    {
+                        context: { active_test: false },
+                    }
+                );
+                if (userData) {
+                    user.partner_id = userData.partner_id[0];
+                }
+            }
+            if (!user.partner_id) {
+                this.notification.add(this.env._t("You can only chat with existing users."), {
+                    type: "warning",
+                });
+                return;
+            }
+            partnerId = user.partner_id;
         }
-        if (!user.partner_id) {
-            this.notification.add(this.env._t("You can only chat with existing users."), {
-                type: "warning",
-            });
+        if (partnerId === this.user.partnerId) {
```

Comment:
You should keep allowing `openChat`/`getChat` to work with current partner.
What I meant in previous comment is that you should disable the specific feature of click on avatar to do `openChat` when it's avatar on current user.

Reply:
Why user should be able to chat with itself but not on click on its avatar?